developer/src/tike/project/Keyman.Developer.System.Project.xmlLdmlProjectFile.pas

```diff
@@ -0,0 +1,184 @@
+{
+ * Keyman is copyright (C) SIL International. MIT License.
+ *
+ * xmlLdmlProjectFile: LDML keyboard files
+}
+unit Keyman.Developer.System.Project.xmlLdmlProjectFile;
+
+interface
+
+uses
+  System.SysUtils,
+  Xml.XMLIntf,
+
+  Keyman.Developer.System.Project.ProjectFile,
+  Keyman.Developer.System.Project.ProjectFiles,
+  Keyman.Developer.System.Project.ProjectFileType,
+  UKeymanTargets;
+
+type
+  TxmlLdmlProjectFile = class;
+
+  TxmlLdmlProjectFile = class(TOpenableProjectFile)
+  private
+    FDebug: Boolean;
+    FHeader_Name: WideString;
+    FTargets: TKeymanTargets;
+    FKVKFileName: string;
+
+    function GetOutputFilename: string;
+    function GetTargetFilename: string;
+    function GetJSTargetFilename: string;
+  protected
+    function GetRelativeOrder: Integer; override;
+    procedure GetFileParameters; override;
+
+    property KVKFileName: string read FKVKFileName;
+    property IsDebug: Boolean read FDebug;
+  public
+    class function IsFileTypeSupported(const Filename: string): Boolean; override;
+    function IsCompilable: Boolean; override;
+    procedure Load(node: IXMLNode); override;   // I4698
+    procedure Save(node: IXMLNode); override;   // I4698
+    procedure LoadState(node: IXMLNode); override;   // I4698
+    procedure SaveState(node: IXMLNode); override;   // I4698
```

Comment:
While I'm not quite following the inheritance and/or polymorphism patterns used here and in ProjectFile.pas (as noticed via commit 1), I can see a pretty strong correlation between the methods.

I'm mostly confused why these `procedure`s are `override` here but `virtual` there; I don't see the inheritance path between the two, and personally would've expected them to be sibling derived classes of a common base.  But those might be small nits and partly due to my inexperience with Delphi in particular.

Reply:
The implementation in ProjectFile.pas is the base class, hence it is virtual. The other classes are the overrides and extend the base functionality.